tty/uart16550/uart16550.c

```diff
@@ -190,7 +190,7 @@ static void uart_ioctl(unsigned int port, msg_t *msg)
 		err = -EINVAL;
 	}
 	else if (req == KIOEN) {
-		if (uart == &uart_common.uarts[UART16550_CONSOLE_USER]) {
+		if ((UART16550_CONSOLE_USER >= 0) && (uart == &uart_common.uarts[UART16550_CONSOLE_USER])) {
```

Comment:
For future reference: many problems stem from using user-provided unsanitized board_config.h, without using an intermediate header for filtering and sanity-checks.